### ISSUE TITLE
Question only: Is this repo suitable for use in production? Also GIFs

### ISSUE THREAD

**Issue by gwhizoftv** [milesj/interweave#224]
Hi All,

We need a supplement to add emojis and GIFs to our React web chat. We are looking at interweave-emoji-picker. Can we include GIFs somehow? Will this repo be maintained any longer ? Many thanks.

**Comment by milesj** [comment]
Yeah this can be used in production. We do in our products at work.

And by GIFs, do you mean just random images that have nothing to do with emojis (I'm assuming like Discords menu)? If so, then no, the emoji picker only supports official emojis in the unicode spec.

**Comment by gwhizoftv** [comment]
@milesj thank you. We do need something like the GIF menu option of discord, along with the emojis. Any suggestions on where to find that additional functionality?

**Comment by milesj** [comment]
@gwhizoftv I'm not aware of one.

It sounds like something custom that will need to be built, since its very unique to your product. With that being said, if you did build it custom, you could still use interweave and emojibase for the emoji part:

- https://emojibase.dev/
- https://interweave.dev/docs/exts/emoji

The emoji picker just pieces those 2 together: https://github.com/milesj/interweave/tree/master/packages/emoji-picker/src

**Comment by gwhizoftv** [comment]
Thank you @milesj . I noticed that Giphy has an SDK, so maybe it would make a companion to your emojis.

**Comment by gwhizoftv** [comment]
Miles,

Thank you for responding so quickly and thoroughly!

We plan to use Interweave for chat message processing, emoji-mart and giphy
API for emojis and GIFs.
For our first step with Interweave, we are thinking to use:

   1.

   For Interweave we want to wrap our text message in Interweave before
   posting it to the chat channel. Interweave will handle some things
   automatically:
   1.

      convert URL’s to clickable links. Currently our chat doesn’t create
      clickables. https://interweave.dev/docs/exts/autolink#urls
      2.

      Convert hashtags to clickable. Let’s use Twitter only
      https://interweave.dev/docs/exts/autolink#urls
      3.

      Escape HTML in case people try to put HTML into the message
      2.

   Create a matcher component that takes our dictionary and replaces with
   ‘****' if matched.
   https://interweave.dev/docs/matchers#creating-a-matcher For testing,
   let’s just use a matcher that can match some basic curse words in english.

In your docs you mention a "factory" matcher. Do you know of any Interweave
matchers and replaceWith() that already handle some language filtering like
the above?

If you see anything wrong or that I misunderstood something, please let me
know!

Once we are done I'm happy to make a pull request if that would be useful
with our changes/additions.
Also to make a donation if you have a patreon, etc.

Many thanks 🙏
Michael

**Comment by milesj** [comment]
I'm not aware of any censoring matchers, but sounds like a good idea.
